auto_submit/lib/service/validation_service.dart

```diff
@@ -194,12 +194,19 @@ class ValidationService {
     }
 
     // If we got to this point it means we are ready to submit the PR.
-    final ProcessMergeResult processed = await processMerge(
+    final ValidationResult processed = await processMerge(
```

Comment:
Shall we create a new validation for it as other validations to simplify the logic? considering here it is being treated as a validation.

Reply:
You mean a new wrapper per se? Yeah I can do that so it is not confused with the ValidationResult.

Reply:
Done. Added a Record!

Reply:
I meant moving the mergeability logic as a `validation` check. Seems we already have a `conflicting` validation, shall we merge the new logic with it?

Reply:
https://github.com/flutter/cocoon/blob/main/auto_submit/lib/validations/conflicting.dart

Reply:
Yeah it technically already is. I thought about this and if you are fine with it I will remove this pre-merge altogether. This was added because there was a request to make a check right before the merge.

https://github.com/flutter/flutter/issues/112018

Otherwise with the previous changes we are essentially doing this if we move it to the validations. We check unknown mergeable and conflicting mergeable already which leaves only the positive state Mergeable.

Reply:
I'd prefer removing code if possible.

Reply:
Yeah, I would move the logic to `conflicting` validation and rename it to say `mergeability`.

With the proposed logic here:
```
switch (mergeableState) {
      case MergeableState.MERGEABLE:
        return (
          result: true,
          action: Action.REMOVE_LABEL,
          message: 'Pull request ${slug.fullName}/$pullRequestNumber is mergeable',
        );
      case MergeableState.UNKNOWN:
        return (
          result: false,
          action: Action.IGNORE_TEMPORARILY,
          message:
              'Mergeability of pull request ${slug.fullName}/$pullRequestNumber could not be determined at time of merge.',
        );
      case MergeableState.CONFLICTING:
        return (
          result: false,
          action: Action.REMOVE_LABEL,
          message: 'Pull request ${slug.fullName}/$pullRequestNumber is not in a mergeable state.',
        );
    }
```
it is enough to cover the the request from https://github.com/flutter/flutter/issues/112018. If it's unknown, then we ignore_temporarily.

Does this make sense to you?

Reply:
Yeah this makes sense. This will be a huge cleanup and will simplify a lot of things.